usecases/config/environment.go

```diff
@@ -254,6 +254,18 @@ func FromEnv(config *Config) error {
 		config.QueryNestedCrossReferenceLimit = DefaultQueryNestedCrossReferenceLimit
 	}
 
+	if v := os.Getenv("QUERY_CROSS_REFERENCE_DEPTH_LIMIT"); v != "" {
+		limit, err := strconv.ParseInt(v, 10, 64)
+		if err != nil {
+			return fmt.Errorf("parse QUERY_CROSS_REFERENCE_DEPTH_LIMIT as int: %w", err)
+		} else if limit <= 0 {
+			limit = math.MaxInt
```

Comment:
if limit < 0 then we set the `limit` to `MaxInt`? shouldn't we always default to `DefaultQueryCrossReferenceDepthLimit`?